### PR TITLE
feat: welcome releasepost experiment

### PR DESCRIPTION
**Pull request by olblak**
A blogpost to introduce an experiment for republishing release notes

### CODE REVIEW

File: content/en/blog/2024/05/welcome_releasepost.md [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+
+To try to answer those questions, I made a small experiment:
+
+Reposting automatically Updatecli release notes to [updatecli.io](https://updatecli.io/changelogs/updatecli), in a way that would allow me to reuse this pipeline for others projects.
+
+Considering [updatecli.io](https://www.updatecli.io) is powered by [HUGO](https://gohugo.io), all I need is a way to automate the synchronization of GitHub release pages with my website's Git repository.
+
+Updatecli already works great to automatically push file updates to git repositories.
+
+Welcome Releasepost
```
Comment (gounthar) [inline content/en/blog/2024/05/welcome_releasepost.md:36]:
```suggestion
Welcome Releasepost.
```

File: content/en/blog/2024/05/welcome_releasepost.md [hunk 2]
```diff
@@ -0,0 +1,424 @@
...
+
+## Releasepost
+
+Releasepost can be downloaded from [github.com/updatecli/releasepost](https://github.com/updatecli/releasepost/releases)
```
Comment (gounthar) [inline content/en/blog/2024/05/welcome_releasepost.md:40]:
```suggestion
Releasepost can be downloaded from [github.com/updatecli/releasepost](https://github.com/updatecli/releasepost/releases).
```

File: content/en/blog/2024/05/welcome_releasepost.md [hunk 3]
```diff
@@ -0,0 +1,424 @@
...
+│   ├── v0.9.3.md
+│   └── v0.9.4.md
+└── index.md
+```
+
+Where the directory `changelogs` contains one file per changelog, and `index.md` a list of links to the various changelogs file.
+
+In theory, it's as simple as that...
+
+Let's see the Releasepost configuration used to generate [www.updatecli.io./changelogs](https://www.updatecli.io/changelogs/updatecli)
```
Comment (gounthar) [inline content/en/blog/2024/05/welcome_releasepost.md:59]:
```suggestion
Let's see the Releasepost configuration used to generate [www.updatecli.io./changelogs](https://www.updatecli.io/changelogs/updatecli):
```

File: content/en/blog/2024/05/welcome_releasepost.md [hunk 4]
```diff
@@ -0,0 +1,424 @@
...
+      repository: udash
+```
+
+sourced from <a href="https://github.com/updatecli/website/blame/master/.releasepost.yaml">releasepost.yaml</a>
+</details>
+
+The most important settings are:
+
+* `spec.owner` and `spec.repository` specify the GitHub repository to monitor for release notes.
+* `dir` specifies where to generate the files
```
Comment (gounthar) [inline content/en/blog/2024/05/welcome_releasepost.md:92]:
```suggestion
* `dir` specifies where to generate the files.
```

File: content/en/blog/2024/05/welcome_releasepost.md [hunk 5]
```diff
@@ -0,0 +1,424 @@
...
+* `formats` specifies the kind of files we want to generate so in this case json and Asciidoctor.
+
+This configuration relies on the environment variable `RELEASEPOST_GITHUB_TOKEN` to set the GitHub credentials used to interact with the API.
+
+[HUGO](https://gohugo.io) relies on specific behaviors like frontmatters to provide page metadata or an index file that differ from other tools like [Docusaurus](https://docusaurus.io/).
+
+I must admit, I was envisioning a simpler configuration that would only contains the GitHub repository to monitor, but as always, the Devil hides in the details.
```
Comment (gounthar) [inline content/en/blog/2024/05/welcome_releasepost.md:99]:
```suggestion
I must admit, I envisioned a simpler configuration that would only contain the GitHub repository to monitor, but as always, the Devil hides in the details.
```

File: content/en/blog/2024/05/welcome_releasepost.md [hunk 6]
```diff
@@ -0,0 +1,424 @@
...
+releasepost --config releasepost.yaml
+```
+
+## Updatecli
+
+Running `releasepost --config` creates the files locally but it's only halfway.
+
+We want to automatically publish them to our git repository.
+
+The next step was to write a Updatecli policy that would run `releasepost` and take care of opening pull requests when needed, such as this [one](https://github.com/updatecli/website/pull/1428).
```
Comment (gounthar) [inline content/en/blog/2024/05/welcome_releasepost.md:116]:
```suggestion
The next step was to write an Updatecli policy that would run `releasepost` and take care of opening pull requests when needed, such as this [one](https://github.com/updatecli/website/pull/1428).
```

File: content/en/blog/2024/05/welcome_releasepost.md [hunk 7]
```diff
@@ -0,0 +1,424 @@
...
+
+## Conclusion
+
+Here we are, to automate release note publishing to a documentation website.
+
+We need three files:
+
+1. A releasepost configuration that specifies the GitHub repository to monitor.
+2. An update-compose.yaml file listing Updatecli policies to enforce.
+3. A values file listing the GitHub repository to monitor for release notes and the GitHub repository to copy those release notes
```
Comment (gounthar) [inline content/en/blog/2024/05/welcome_releasepost.md:391]:
```suggestion
3. A values file listing the GitHub repository to monitor for release notes and the GitHub repository to copy those release notes.
```

File: content/en/blog/2024/05/welcome_releasepost.md [hunk 8]
```diff
@@ -0,0 +1,424 @@
...
+4. A GitHub action workflow to execute Updatecli periodically.
+
+To answer my initial questions.
+
+_What if I could visualize Updatecli changelogs from [www.updatecli.io](https://www.updatecli.io/changelogs/updatecli)?_
+
+It's working great for a few months now, and I am not looking back.
```
Comment (gounthar) [inline content/en/blog/2024/05/welcome_releasepost.md:398]:
```suggestion
It's been working great for a few months now, and I am not looking back.
```

File: content/en/blog/2024/05/welcome_releasepost.md [hunk 9]
```diff
@@ -0,0 +1,424 @@
...
+
+_How many people read release notes?_
+
+Traffic show me that people/crawler do care.
+
+_Does it even make sense to spend effort maintaining those release notes?_
+
+I do think so, manually crafted release notes are by far the best ones but they are also the most time-consuming. It's difficult to maintain them over time for every project so automatically generated release notes are a good alternative.
+
+I wish more project consider release notes as a critical component of their releases.
```
Comment (gounthar) [inline content/en/blog/2024/05/welcome_releasepost.md:408]:
```suggestion
I wish more projects consider release notes as a critical component of their releases.
```

File: content/en/blog/2024/05/welcome_releasepost.md [hunk 10]
```diff
@@ -0,0 +1,424 @@
...
+
+## Next steps
+
+This project was built with flexibility in mind, so we could definitely envisioned building plugins for NPM, Gitlab releases, etc.
```
Comment (gounthar) [inline content/en/blog/2024/05/welcome_releasepost.md:412]:
```suggestion
This project was built with flexibility in mind, so we could definitely envision building plugins for NPM, Gitlab releases, etc.
```

File: content/en/blog/2024/05/welcome_releasepost.md [hunk 11]
```diff
@@ -0,0 +1,424 @@
...
+
+In the near by future, I am considering documenting how we automate release note publishing on Docusaurus website leveraging the versioning feature.
```
Comment (gounthar) [inline content/en/blog/2024/05/welcome_releasepost.md:414]:
```suggestion
In the near future, I am considering documenting how we automate release notes publishing on Docusaurus website by leveraging the versioning feature.
```

File: content/en/blog/2024/05/welcome_releasepost.md [hunk 12]
```diff
@@ -0,0 +1,424 @@
...
+In that scenario, we only need release notes associated to the versioned website.
+
+Example on [fleet.rancher.io/0.9/changelogs](https://fleet.rancher.io/0.9/changelogs/)
+
+All of this don't replace the need to write good release notes, we just try to make them more visible.
```
Comment (gounthar) [inline content/en/blog/2024/05/welcome_releasepost.md:419]:
```suggestion
All of this doesn't replace the need to write good release notes, we just try to make them more visible.
```

**Review by gounthar — APPROVED** [review]

**Comment by olblak** [comment]
Thanks @gounthar for the reviews

File: content/en/blog/2024/05/welcome_releasepost.md [hunk 13]
```diff
@@ -0,0 +1,424 @@
...
+        indexfilename: _index
+         indexfrontmatters: |
+           ---
+          title: "Changelogs"
+          ---
+      - extension: json
+        indexfilename: _index
+    spec:
+      owner: updatecli
+      repository: udash
```
Comment (olblak) [inline content/en/blog/2024/05/welcome_releasepost.md:83]:
```suggestion
      repository: updatecli
```

File: content/en/blog/2024/05/welcome_releasepost.md [hunk 14]
```diff
@@ -0,0 +1,424 @@
...
+
+The most important settings are:
+
+* `spec.owner` and `spec.repository` specify the GitHub repository to monitor for release notes.
+* `dir` specifies where to generate the files.
+* `formats` specifies the kind of files we want to generate so in this case json and Asciidoctor.
+
+This configuration relies on the environment variable `RELEASEPOST_GITHUB_TOKEN` to set the GitHub credentials used to interact with the API.
+
+[HUGO](https://gohugo.io) relies on specific behaviors like frontmatters to provide page metadata or an index file that differ from other tools like [Docusaurus](https://docusaurus.io/).
```
Comment (olblak) [inline content/en/blog/2024/05/welcome_releasepost.md:97]:
```suggestion
[HUGO](https://gohugo.io) relies on specific behaviors like frontmatters to provide page metadata or an index file name that differ from other tools like [Docusaurus](https://docusaurus.io/).
```

File: content/en/blog/2024/05/welcome_releasepost.md [hunk 15]
```diff
@@ -0,0 +1,424 @@
...
+
+I must admit, I envisioned a simpler configuration that would only contain the GitHub repository to monitor, but as always, the Devil hides in the details.
+
+NOTE: More customization is available such as version filtering or the credentials to use but let's keep that for later.
+
+You can try releasepost by running:
+
+```bash
+export RELEASEPOST_GITHUB_TOKEN=ghp_xxx
```
Comment (olblak) [inline content/en/blog/2024/05/welcome_releasepost.md:106]:
```suggestion
export RELEASEPOST_GITHUB_TOKEN=<insert here a GitHub personnal token>
```

File: content/en/blog/2024/05/welcome_releasepost.md [hunk 16]
```diff
@@ -0,0 +1,424 @@
...
+
+Running `releasepost --config` creates the files locally but it's only halfway.
+
+We want to automatically publish them to our git repository.
+
+The next step was to write an Updatecli policy that would run `releasepost` and take care of opening pull requests when needed, such as this [one](https://github.com/updatecli/website/pull/1428).
+
+### Updatecli Policy
+
+On the Updatecli project, we are using this policy:
```
Comment (olblak) [inline content/en/blog/2024/05/welcome_releasepost.md:120]:
```suggestion
On the Updatecli project, we are using this Updatecli policy:
```

File: content/en/blog/2024/05/welcome_releasepost.md [hunk 17]
```diff
@@ -0,0 +1,424 @@
...
+                - name: RELEASEPOST_GITHUB_TOKEN
+                - name: PATH
+version: 0.77.0
+```
+
+</details>
+
+#### To visualize what this pipeline would change
+
+<details><summary><code>updatecli diff ghcr.io/updatecli/policies/releasepost/releasepost:0.1.0</code></summary>
```
Comment (olblak) [inline content/en/blog/2024/05/welcome_releasepost.md:173]:
```suggestion
<details><summary><code>updatecli diff ghcr.io/updatecli/policies/releasepost/releasepost:0.1.0</code></summary>

Please note that to make the following output smaller, the releasepost configuration was modified to monitor the GitHub repositroy `updatecli/udash` which contains far less releases than `updatecli/updatecli`
```

File: content/en/blog/2024/05/welcome_releasepost.md [hunk 18]
```diff
@@ -0,0 +1,426 @@
...
+[HUGO](https://gohugo.io) relies on specific behaviors like front-matters to provide page metadata or an index file name that differs from other tools like [Docusaurus](https://docusaurus.io/).
+
+I must admit, I envisioned a simpler configuration that would only contain the GitHub repository to monitor, but as always, the Devil hides in the details.
+
+NOTE: More customization is available such as version filtering or the credentials to use but let's keep that for later.
+
+You can try releasepost by running:
+
+```bash
+export RELEASEPOST_GITHUB_TOKEN=<insert here a GitHub personnal token>
```
Comment (olblak) [inline content/en/blog/2024/05/welcome_releasepost.md:106]:
```suggestion
export RELEASEPOST_GITHUB_TOKEN=<insert here a GitHub personal token>
```

**Pull request merged**